src/components/TimeRestrictionsTable.tsx

```diff
@@ -8,77 +8,138 @@ interface TimeRestrictionsTableProps {
-                    <div className="govuk-grid-column-one-quarter">
-                        <FormGroupWrapper errors={errors} errorId={`end-time-${chosenDay}`}>
-                            <FormElementWrapper
-                                errors={errors}
-                                errorId={`end-time-${chosenDay}`}
-                                errorClass="govuk-input--error"
-                            >
-                                <>
-                                    <label
-                                        className={`govuk-label ${index === 0 ? '' : 'govuk-visually-hidden'}`}
-                                        htmlFor={`end-time-${chosenDay}`}
-                                    >
-                                        End time <span className="govuk-visually-hidden">for {chosenDay}</span>
-                                    </label>
+                    <div>
+                        <FormGroupWrapper errors={errors} errorId={`end-time-${chosenDay}-${i}`}>
+                            <>
+                                <label className="govuk-label" htmlFor={`end-time-${chosenDay}`}>
+                                    End time <span className="govuk-visually-hidden">for {chosenDay}</span>
+                                </label>
+                                <FormElementWrapper
+                                    errors={errors}
+                                    errorId={`end-time-${chosenDay}-${i}`}
+                                    errorClass="govuk-input--error"
+                                >
                                     <input
-                                        className="govuk-input govuk-input--width-4"
-                                        id={`end-time-${chosenDay}`}
+                                        className="govuk-input govuk-input--width-5"
+                                        id={`end-time-${chosenDay}-${i}`}
                                         name={`endTime${chosenDay}`}
                                         aria-describedby="time-restrictions-hint"
                                         type="text"
-                                        defaultValue={findCorrectDefaultValue(endTimeInputs, chosenDay)}
+                                        defaultValue={findCorrectDefaultValue(endInputsForDay, chosenDay, i)}
                                     />
-                                </>
-                            </FormElementWrapper>
+                                </FormElementWrapper>
+                            </>
                         </FormGroupWrapper>
                     </div>
-                </fieldset>
+                </div>,
+            );
+        }
+
+        return rows;
+    };
+
+    return (
+        <>
+            {chosenDays.map(chosenDay => (
+                <>
+                    <fieldset className="govuk-fieldset flex-container time-restrictions-table" key={chosenDay}>
```

Comment:
The key here will need to be moved up to the fragment which means you'll need to use the full syntax for the fragment as well `<Fragment key={chosenDay}>...</Fragment>